resources/startup.sh

```diff
@@ -96,151 +90,207 @@ function install_plugins(){
 # installs or upgrades the given plugin
 function install_plugin(){
   NAME="${1}"
-  SOURCE="${PLUGIN_STORE}/${NAME}"
+  SOURCE="${DEFAULT_PLUGIN_DIRECTORY}/${NAME}"
   TARGET="${PLUGIN_DIRECTORY}/${NAME}"
 
   if [ ! -d "${SOURCE}" ]; then
-    exit 1
+    echo "ERROR: ${SOURCE} is not a directory. Skipping this plugin..."
+    return
   fi
 
-  echo "install plugin ${NAME}"
   if [ -d "${TARGET}" ]; then
-    rm -rf "${TARGET}"
+    echo "Plugin ${NAME} already exists. Skip restoring the plugin..."
+    return
   fi
 
+  echo "install plugin ${NAME}"
   cp -rf "${SOURCE}" "${TARGET}"
 }
 
-echo "render config.ru template"
-render_config_ru_template
+function installBundledPluginsIfNecessary() {
+  PLUGINS=$(ls "${DEFAULT_PLUGIN_DIRECTORY}")
+
+  for PLUGIN_PACKAGE in ${PLUGINS}; do
+    install_plugin "${PLUGIN_PACKAGE}"
+  done
+}
+
+function installPluginGems() {
+  # Install missing gems if new plugins are going to be installed.
+  # Otherwise bundle will detect that there are no changes and thus no new gems are needed
+  echo "install missing gems ..."
+  
+  rakeExitCode=0
+  RAILS_ENV="${RAILS_ENV}" REDMINE_LANG="${REDMINE_LANG}" bundle install || rakeExitCode=$?
+  if [[ ${rakeExitCode} -ne 0 ]]; then
+    echo "ERROR: Rake bund installed returned with an error during the gem installation for new plugins"
+    sleep 300
```

Comment:
Why sleep here?

Reply:
any non-zero exit leads to a container restart because of `set -o errexit -o nounset -o pipefail`.
Staying in a crash loop for a day would produce a heavy log load with no further added value. So it is good dogu practice to sleep for a while and then restart. The Dogu in turn would be perceived as down in both scenarios.